src/order/interval.lean

```diff
@@ -0,0 +1,422 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.set.intervals.basic
+import data.set.lattice
+
+/-!
+# Order intervals
+
+This file defines (nonempty) intervals in an order. This is a prototype for interval arithmetic.
```

Comment:
I think it's also worth saying that this treats intervals as `Icc`, and some applications may instead prefer to work with `Ico` or `Ioc`